packages/presentation/presentation_publishing/interfaces/publishes_unsaved_changes.ts

```diff
@@ -11,7 +11,7 @@ import { PublishingSubject } from '../publishing_subject';
 
 export interface PublishesUnsavedChanges<Runtime extends object = object> {
   unsavedChanges: PublishingSubject<Partial<Runtime> | undefined>;
-  resetUnsavedChanges: () => void;
+  resetUnsavedChanges: () => boolean;
```

Comment:
Do you think we should remove this typing after the architectural fix is in? I think this should be temporary because in theory, the `reset` function shouldn't be capable of failing.